Installation.md

```diff
@@ -0,0 +1,267 @@
+```
+After download finishes extract the files and create symlinks
+```
+$ tar -xvjf $PHANTOM_JS.tar.bz2
+$ sudo mv $PHANTOM_JS /usr/local/share
+$ sudo ln -s /usr/local/share/$PHANTOM_JS/bin/phantomjs /usr/local/bin
+$ rm $PHANTOM_JS.tar.bz2
+```
+Check that PhantomJS is installed correctly
+```
+$ phantomjs --version
+```
+
+### Install ruby via RVM ###
+#### Install RVM as multi-user install ####
+
+```
+$ gpg --keyserver hkp://keys.gnupg.net --recv-keys 409B6B1796C275462A1703113804BB82D39DC0E3
+$ \curl -L https://get.rvm.io | sudo bash -s stable
+```
+
+After installation finishes add new user and setup groups
+
+```
+$ sudo adduser tmc
+$ sudo addgroup rvm
+$ sudo usermod -a -G rvm tmc
+$ sudo /usr/local/rvm/bin/rvm install $RUBY_VERSION
+$ sudo /usr/local/rvm/bin/rvm use $RUBY_VERSION --default
+```
+
+Setup ruby path
+
+```
+$ echo 'export PATH=$PATH:/usr/local/rvm/rubies/ruby-$RUBY_VERSION/bin' | sudo tee -a etc/profile.d/ruby.sh
+```
+
+#### Switch to tmc user and login their shell ####
+
+```
+$ sudo su tmc
+$ bash -l
+$ cd
+```
+
+#### Import public key for rvm installation ####
+
+```
+$ command curl -sSL https://rvm.io/mpapis.asc | gpg --import -
+$ rvm get stable
+```
+
+#### Setup user environment variables ####
+
+```
+$ echo "export GEM_HOME=$(ruby -e 'print Gem.user_dir')" >> ~/.bashrc
+$ echo "export PATH=$(echo $PATH):$GEM_HOME/bin" >> ~/.bashrc
+$ export GEM_HOME=$(ruby -e 'print Gem.user_dir')
+$ export PATH=$PATH:$GEM_HOME/bin
+```
+
+### PostgreSQL ###
+#### Create postgres user ####
+
+Run following command and when prompted give **tmc** as a password, if you want to use another username or stronger password change them into **config/database.yml**.
+
+```
+$ createuser tmc -s -P
+```
+
+If you cannot run above command, you can also create the user manually
+
+```
+$ sudo -s
+$ su - postgres
+$ psql
+    CREATE USER tmc WITH SUPERUSER PASSWORD 'tmc';
+```
+
+#### Set PostgreSQL to authenticate with md5 instead of local ####
+
+Locate the **pg_hba.conf** file with `locate pg_hba.conf`, the file should be located */etc/postgresql/<version number>/main/pg_hba.conf*. Open the file with your preferred text-editor, you need root privileges to edit the file. Around line 90 change
+```
+# "local" is for Unix domain socket connections only
+local   all             all                                     peer
+```
+to
+```
+# "local" is for Unix domain socket connections only
+local   all             all                                     md5
+```
+
+### TMC-server installation ###
+#### Clone the TMC repository ####
+
+```
+$ cd tmc-server
+$ git clone https://github.com/testmycode/tmc-server.git
+```
+:exclamation: If you are not using a github account, replace the repository submodule URLs with a HTTPS URL in .git/config eg. https://github.com/testmycode/tmc-checkstyle-runner.git
+```
+$ git submodule update --init --recursive
+$ gem install bundler && bundle install
+$ cp config/site.defaults.yml confg/site.yml
```

Comment:
Rather than copying the entire file, mention which options one needs to change (are there even any for a dev setup). 

Note that the entire file doesn't need to be copied. Settings not in `site.yml` will be looked up from `site.defaults.yml`.